File: src/main/java/kdh/boardproject/dto/category/CategoryDto.java
package kdh.boardproject.dto.category;

import kdh.boardproject.dto.user.UserRequestDto;
import kdh.boardproject.entity.Category;
import kdh.boardproject.entity.User;
import lombok.*;

@Data
public class CategoryDto {
    private Long categoryIdx;
    private String categoryName;
    private String description;
    private UserRequestDto user;

    public CategoryDto(Category category){
        this.categoryIdx = category.getIdx();
        this.categoryName = category.getCategoryName();
        this.description = category.getDescription();
        this.user = new UserRequestDto(category.getUser());
    }
}

File: src/main/java/kdh/boardproject/dto/category/CategoryDto.kt
package kdh.boardproject.dto.category

import kdh.boardproject.dto.user.UserRequestDto
import kdh.boardproject.entity.Category
import lombok.Data

@Data
class CategoryDto(category: Category) {
    var categoryIdx: Long?
    var categoryName: String
    var description: String
    var user: UserRequestDto

    init {
        categoryIdx = category.idx
        categoryName = category.categoryName
        description = category.description
        user = UserRequestDto(category.user)
    }
}